tools/src/main/java/org/apache/kafka/tools/ManifestWorkspace.java

```diff
@@ -0,0 +1,573 @@
+        }
+    }
+
+    /**
+     * Rewrite a file, creating it and any parent directories if it does not exist, deleting it if specified
+     *
+     * @param dryRun              True if the rewrite should be applied, false if it should be simulated.
+     * @param filePath            Path to file which should be rewritten.
+     * @param elements            Class names of plugins which should appear in the manifest
+     */
+    private void rewriteManifestFile(boolean dryRun, Path filePath, Set<String> elements) throws IOException, TerseException {
+        Objects.requireNonNull(filePath, "filePath must be non-null");
+        Objects.requireNonNull(elements, "elements must be non-null");
+        Path writableFile = getWritablePath(dryRun, filePath);
+        if (elements.isEmpty()) {
+            if (Files.exists(filePath)) {
+                out.printf("\tDelete\t%s%n", filePath);
+                if (!dryRun) {
+                    Files.delete(writableFile);
+                }
+            }
+            // TODO: delete parent directories if empty up to the containing directory
+        } else {
+            Path parent = filePath.getParent();
+            if (parent != null) {
+                if (Files.exists(parent)) {
+                    if (!Files.isWritable(parent)) {
+                        throw new TerseException(parent + " is not writable");
+                    }
+                } else {
+                    out.printf("\tCreate\t%s%n", parent);
+                    if (!dryRun) {
+                        Files.createDirectories(parent);
+                    }
+                }
+            }
+            if (!Files.exists(filePath)) {
+                out.printf("\tCreate\t%s%n", filePath);
+            }
+            out.printf("\tWrite\t%s with content %s%n", filePath, elements);
+            if (!dryRun) {
+                try (OutputStream stream = new BufferedOutputStream(Files.newOutputStream(
+                        writableFile,
+                        StandardOpenOption.CREATE,
+                        StandardOpenOption.WRITE,
+                        StandardOpenOption.TRUNCATE_EXISTING
+                ))) {
+                    byte[] newline = System.lineSeparator().getBytes(StandardCharsets.UTF_8);
+                    String header = "# Generated by connect-plugin-path.sh";
+                    stream.write(header.getBytes(StandardCharsets.UTF_8));
+                    stream.write(newline);
+                    for (String element : elements) {
+                        stream.write(element.getBytes(StandardCharsets.UTF_8));
+                        stream.write(newline);
+                    }
+                }
+            }
+        }
+    }
+
+    /**
+     * Get a path which is always writable
+     * @param dryRun If true, substitute a temporary file instead of the real file on disk.
+     * @param path Path which must be writable, and may be different from the input path
```

Comment:
Is this part correct? I'm unsure what the difference between the `path` parameter and the "input path" is. Seems like part of this description may have been referring to the return value instead?